codalab/lib/bundle_store.py

```diff
@@ -383,3 +395,138 @@ def cleanup(self, uuid, dry_run):
         if not dry_run:
             path_util.remove(absolute_path)
 
+
+    def health_check(self, model, force=False, compute_data_hash=False, repair_hashes=False):
+        """
+        MultiDiskBundleStore.health_check(): In the MultiDiskBundleStore, bundle contents are stored on disk, and
+        occasionally the disk gets out of sync with the database, in which case we make repairs in the following ways:
+
+            1. Deletes bundles with corresponding UUID not in the database.
+            3. Deletes any files not beginning with UUID string.
+            4. For each bundle marked READY or FAILED, ensure that its dependencies are not located in the bundle
+               directory. If they are then delete the dependencies.
+            5. For bundle <UUID> marked READY or FAILED, <UUID>.cid or <UUID>.status, or the <UUID>(-internal).sh files
+               should not exist.
+        |force|: Perform any destructive operations on the bundle store the health check determines are necessary. False by default
+        |compute_data_hash|: If True, compute the data_hash for every single bundle ourselves and see if it's consistent with what's in
+                             the database. False by default.
+        """
+        UUID_REGEX = re.compile(r'^(0x[0-9a-z]{32})')
```

Comment:
UUID_REGEX in spec_util assumes that the entire string is the UUID, here we just wanted to do regex matching to see if the filename begins with a full uuid. I can use spec_util.UUID_STR to build this regex though